pkg/sql/resolver.go

```diff
@@ -413,6 +413,50 @@ func (p *planner) ResolveDescriptorForPrivilegeSpecifier(
 			return nil, err
 		}
 		return table, nil
+	} else if specifier.SequenceName != nil || specifier.SequenceOID != nil {
```

Comment:
sequences are treated the same as tables and as such most of this logic is the same as above.

maybe we should change `HasPrivilegeSpecifier` to 

```
	// Table privilege
	TableName *string
	TableOID  *oid.Oid
  IsSequence bool
```

for the different sequence check?
also, what happens in PG when you do `has_table_privilege` on a sequence?

Reply:
> also, what happens in PG when you do has_table_privilege on a sequence?

ah, looks like it works as well!

Reply:
Yeah I saw we had a test that expects that to work, but it seems like it shouldn't.  

Should I set up postgres and see what it does and match this behavior?

Reply:
I'll add `IsSequence bool` though to reduce code duplication.

Reply:
always have PG setup; we mostly imitate their behaviour :)

it does work in PG so we should do the same.